db/db_test_util.cc

```diff
@@ -496,6 +496,13 @@ Status DBTestBase::ReadOnlyReopen(const Options& options) {
 
 Status DBTestBase::TryReopen(const Options& options) {
   Close();
+  last_options_.table_factory.reset();
```

Comment:
Sure. I can do a separate patch. But I already tried the destruction approach and it causes other issues. I did not have time to look into it but apparently it would also destruct some objects that will be reused later.